src/components/dialogs/network-modifications/coupling-device/coupling-device-dialog.jsx

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright (c) 2025, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { BUS_BAR_SECTION_ID1, BUS_BAR_SECTION_ID2 } from 'components/utils/field-constants';
+import { useCallback, useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { FetchStatus } from '../../../../services/utils.js';
+import { EquipmentIdSelector } from '../../equipment-id/equipment-id-selector.js';
+import { EQUIPMENT_TYPES } from '../../../utils/equipment-types.js';
+import { ModificationDialog } from '../../commons/modificationDialog.js';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useOpenShortWaitFetching } from '../../commons/handle-modification-form.js';
+import { FORM_LOADING_DELAY } from '../../../network/constants.js';
+import { createCouplingDevice } from '../../../../services/study/network-modifications.js';
+import { CustomFormProvider, useSnackMessage } from '@gridsuite/commons-ui';
+import yup from '../../../utils/yup-config.js';
+import { fetchBusesOrBusbarSectionsForVoltageLevel } from '../../../../services/study/network.js';
+import { CouplingDeviceForm } from './coupling-device-form.jsx';
+import { useIntl } from 'react-intl';
+import { isNodeBuilt } from '../../../graph/util/model-functions';
+
+const emptyFormData = {
+    [BUS_BAR_SECTION_ID1]: null,
+    [BUS_BAR_SECTION_ID2]: null,
+};
+const formSchema = yup.object().shape({
+    [BUS_BAR_SECTION_ID1]: yup.object().required().nullable(),
+    [BUS_BAR_SECTION_ID2]: yup.object().required().nullable(),
+});
+export const CouplingDeviceDialog = ({
```

Comment:
```suggestion
export const CouplingDeviceCreationDialog = ({
```
It's better to be homogeneous with the backend modification name